iscsi/iscsi.go

```diff
@@ -33,20 +34,18 @@ type iscsiSession struct {
 
 //Connector provides a struct to hold all of the needed parameters to make our iscsi connection
 type Connector struct {
-	VolumeName       string
-	TargetIqn        string
-	TargetPortals    []string
-	Port             string
-	Lun              int32
-	AuthType         string
-	DiscoverySecrets Secrets
-	SessionSecrets   Secrets
-	Interface        string
-	Multipath        bool
-	// Number of time we try to obtain a device path, with CheckInterval seconds between each attempt
-	RetryCount int32
-	// Time (in seconds) between login attempts
-	CheckInterval int32
+	VolumeName       string   `json:"volume_name"`
+	TargetIqn        string   `json:"target_iqn"`
+	TargetPortals    []string `json:"target_portals"`
+	Port             string   `json:"port"`
+	Lun              int32    `json:"lun"`
+	AuthType         string   `json:"auth_type"`
+	DiscoverySecrets Secrets  `json:"discovery_secrets"`
```

Comment:
Make sure there's no logging of these secrets anywhere

Reply:
Good catch, now that you mention it, there is a potential for that (unrelated to this patch) if debug is enabled and the update DB call fails we log the arguments to iscsiadm (which would include the secrets we're trying to set).

I'll work on a patch to scrub those, but that should be independent of this PR I think.